### ISSUE TITLE
Current TODO list

### ISSUE THREAD

**Issue by Gurrman375** [kay-el-zed/HaloMCC-DiscordRPC#5]
I thought a TODO list would be great for the future of this project as a sort of project tracker.
- [ ] Big TODO
  - [ ] Clean up code. (pushing back for now but is a high priority)
  - [x] Add session timer tracking, i.e. How long a multiplayer session has run for (Chcking this because there isn't a way to get the current time even with the steam version. Also the elapsed time typically shows how long you have been playing it.)
  - [ ] Reduce debugging output. (Option to select a heavier debug)
  - [ ] Possibly perform better detection for Halo MCC on PC by checking for processes (Finding the steam id might help?)
- [ ] Extra TODO
  - [ ] Better Gui (UI is available but working on GUI)
  - [ ] Options (Choose what you want to show on your profile)
  - [x] Game Detection (Works for every game except Halo 3 Odst and all Firefights)
  - [x] Join (Currently working with steam. Windows and Xbox might not be possible.)
  - [ ] Add better checks for if the user is playing on either Xbox or PC. (Doing this is a low priority atm.)
--------------------------------------------------------------------------------------------------------------
I hope to get the big TODO finished. On year has past and I have hardly gone through this list.

**Comment by kay-el-zed** [comment]
That all sounds like a great plan to me. A couple notes while I'm thinking about them:

- I think adding Xbox checking shouldn't be too big of a hurdle. I believe it should be a matter of adding something similar to `ActiveTitles['type'] == "Win32` for checking if playing on Xbox.
- I'm all for adding better detecting, but would prefer nothing that's invasive to the game's process. I feel as though it'll be likely to trip the anti-cheat if we start poking around too much with the executable. It doesn't seem that 343 has an official API for MCC available to the public, so we might not be able to do more than just scanning running processes to look for `MCC-Win64-Shipping.exe`.

**Comment by Gurrman375** [comment]
I prefer running away from the game process but unless 343 release it's api timed matches would have to go in the modded version. And that's not even mentioning the windows store version.

**Comment by kay-el-zed** [comment]
Makes sense. I completely forgot about the modded version - poking around in non-EAC mode would be fair game. 

I have no idea what's up with the Windows Store version - I know there's a ton of complexity due to UWP sandboxing that might make things messy.

**Comment by Gurrman375** [comment]
It's very odd with the window store version because the executable is there and usable but might have it's own anti cheat system

**Comment by Gurrman375** [comment]
Make sure you put the exe as the next release.

**Comment by Gurrman375** [comment]
Still working on this project. Game Detection is taking much longer than I thought it would. Also working on some GUI builds. Still in very alpha state so till next time.

**Comment by Gurrman375** [comment]
Hey Everyone.
So in the past month or two I have been very quiet. There are reasons for that. I wanted to address some issues with the program and why it's taking so long to complete.
First, my computer died. : (
I haven't been able to work since then so all I could do is watch as the progress made is slowly dying.
Second, I have finished gathering more images and working on making sure these reflect the current and future builds of MCC. This has taken some time and consideration, especially with ODST.
Finally, I decided to rework most of the code. This issue had to deal with the activity message showing up and the fact that there was no method to switch between rpc clients.
I will soon post a beta build (may look like an alpha build) that should address and fix these issues.
Till then this is me signing off.

If you want to message me, my DM's are open. Discord: Gurr#8106
